hw/scripts/common.sh

```diff
@@ -76,3 +76,27 @@ parse_extra_jtag_cmd() {
     echo $NEW_EXTRA_JTAG_CMD
     EXTRA_JTAG_CMD=$NEW_EXTRA_JTAG_CMD
 }
+
+# Try to detect connnected programmers
+detect_programmer() {
+
+    # scan USB for well-known VID:PID
+    USB_DEV=$(ls /sys/bus/hid/devices)
```

Comment:
Are you sure that _stlink_ and _jlink_ are **HID** devices.
It does not show up on my Linux (for Windows there is no such folder but we can leave it for now).
`lsusb` on the other hand does show instances of stlink, jlink and dap.
Maybe command is valid for Mac?

Reply:
I've trusted on sysfs too much :angel: _lsusb_ command is the right way. Reviewing now... Thanks for your comments :+1: